cli/src/cli/chain_log.rs

```diff
@@ -0,0 +1,72 @@
+use crate::error::DefaultResult;
+use colored::*;
+use holochain_cas_implementations::cas::file::FilesystemStorage;
+use holochain_core::agent::{
+    chain_store::ChainStore,
+    state::{AgentState, AgentStateSnapshot},
+};
+use holochain_core_types::{chain_header::ChainHeader, entry::Entry};
+use std::{convert::TryFrom, fs, path::PathBuf};
+
+// TODO: use system-agnostic default path
+const DEFAULT_CHAIN_PATH: &str = "/home/michael/.holochain/holo/storage/";
+
+pub fn chain_log(storage_path: Option<PathBuf>, instance_id: String) -> DefaultResult<()> {
+    let storage_path = storage_path.unwrap_or_else(|| PathBuf::new().join(DEFAULT_CHAIN_PATH));
+    let cas_path = storage_path.join(instance_id).join("cas");
+    let snapshot_path = cas_path.join("AgentState.txt");
+    let snapshot_json = fs::read_to_string(snapshot_path)?;
```

Comment:
Since you're creating a `FilesystemStorage` below anyway, you could get the snapshot also through that. It's *address* is `"AgentState"`. ;)

Reply:
Ahh cool :)